streaming/src/main/scala/org/apache/spark/streaming/dstream/TransformedDStream.scala

```diff
@@ -39,7 +39,10 @@ class TransformedDStream[U: ClassTag] (
   override def slideDuration: Duration = parents.head.slideDuration
 
   override def compute(validTime: Time): Option[RDD[U]] = {
-    val parentRDDs = parents.map(_.getOrCompute(validTime).orNull).toSeq
+    val parentRDDs = parents.map { parent => parent.getOrCompute(validTime).getOrElse(
```

Comment:
You've probably thought through this, so I'm mostly asking: do we want to assume the parent can be computed? in some places `null` or `None` does correctly mean "can't be computed now". That may not be here. I'm making sure we don't 'fix' the wrong part.

PS you can still retain the compact `_.getOrCompute` syntax here.